packages/web/src/pages/upload-page/components/FinishPageNew.tsx

```diff
@@ -120,11 +136,57 @@ export const FinishPageNew = (props: FinishPageProps) => {
     onContinue()
   }, [onContinue])
 
-  const handleVisitProfileClick = useCallback(() => {
-    if (user) {
-      dispatch(push(profilePage(user.handle)))
+  const visitButtonText = useMemo(() => {
+    switch (uploadType) {
+      case UploadType.INDIVIDUAL_TRACK:
+        return messages.visitTrack
+      case UploadType.ALBUM:
+        return messages.visitAlbum
+      case UploadType.PLAYLIST:
+        return messages.visitPlaylist
+      default:
+        if (!upload.tracks || upload.tracks.length > 1) {
+          return messages.visitProfile
+        } else {
+          return messages.visitTrack
+        }
+    }
+  }, [upload.tracks, uploadType])
+
+  const handleVisitClick = useCallback(() => {
+    let route = ''
+    switch (uploadType) {
+      case UploadType.INDIVIDUAL_TRACK:
+        route = upload.tracks![0].metadata.permalink
+        break
+      case UploadType.ALBUM:
+      case UploadType.PLAYLIST:
+        route = collectionPage(
+          user!.handle,
+          upload.metadata?.playlist_name,
+          upload.completionId,
+          null,
+          uploadType === UploadType.ALBUM
+        )
+        break
+      default:
+        if (!upload.tracks || upload.tracks.length > 1) {
+          route = profilePage(user!.handle)
+        } else {
+          route = upload.tracks![0].metadata.permalink
+        }
     }
-  }, [dispatch, user])
+
+    dispatch(push(route))
+    dispatch(make(Name.TRACK_UPLOAD_VIEW_TRACK_PAGE, { uploadType }))
+  }, [
+    dispatch,
+    upload.completionId,
+    upload.metadata?.playlist_name,
+    upload.tracks,
+    uploadType,
+    user
+  ])
 
```

Comment:
Can we make this all use a link element instead?

Reply:
check if harmony button allows "as" prop to set it as a Link component. If not lmk

Reply:
I wrapped the button with the Link component, not sure if thats a no-no, but it seems to work fine